esp-hal/src/uart.rs

```diff
@@ -1317,6 +1313,7 @@ impl Uart<'_, Blocking> {
     }
 }
 
+#[instability::unstable]
 impl<Dm> ufmt_write::uWrite for Uart<'_, Dm>
```

Comment:
Do we actually need `uWrite` for something?

Reply:
It was originally requested by a community member IIRC, presumably there are people that use it. Not sure if we want to keep it or not, don't have strong opinions here.

Reply:
I'm wondering what we want to do w.r.t. these trait dependencies. There is no strong reason for not supporting them, but they are also not needed by everyone. We also may want to support multiple versions, or maybe not. I think we might want to come up with a policy regarding them, and apply that to our code.

Hiding every one of these behind "unstable" seems a bit off to me (i.e. forcing users to blanket-enable all unstable code for a trait like uWrite may be a bit too heavy handed), but also a `dependency+version` feature (i.e. for rng-core 0.6 and 0.9, as it's simple enough to maybe support both) may be a bit too granular. And with that, we would essentially commit to supporting a particular version for the lifetime of esp-hal 1.0.

Reply:
I agree that gating this all behind the `unstable` feature is probably a bit of a ham-fisted approach, however I'm also not super excited about the prospect of potentially accumulating a bunch of different versions of various dependencies, each with their own corresponding features. Will have to think about this a bit, I guess.

Reply:
I think the intention of having this unstable is that `ufmt` is 0.2.0

Reply:
mhhh - now we implement the ufmt 0.2.0 trait for stable 🤔 

even if we had a feature like `ufmt` (which we had at some point IIRC) we should have the impl gated by both features ("unstable" + "ufmt") as long as it's <1.0.0

Reply:
One option is to expand the `unstable` feature to have `unstable-ufmt` etc. I don't think we want to have too many of these, but maybe for dependencies it makes sense?

Reply:
Hm, I'm not sure, we were removing the features some time ago and now we want to start adding them again? 🤔

Reply:
https://github.com/esp-rs/esp-hal/pull/34/files#diff-0f5c4630136975227678ae75d645b76f8dfdc8aab233418177130443e9842e84

we had the feature initially - one day it got removed 🤷‍♂️

Reply:
> we spend more time talking/thinking about it than it would take to remove and re-add it 😆

This is a good point but ufmt isn't the only such dependency we have. Are we also considering removing rand-core? embassy-embedded-hal? embassy-usb-*? log? Some of these are optional already through different means, but they are unstable, so if they release a breaking change (e.g. log 0.5), that will affect us.

My point is, we need to figure out these dependencies, and what we do with them. We can remove a specific example, but the issue doesn't go away just from that.

Reply:
+1 - do we have an issue for the general discussion? I thought we had one but I'm unable to find it 🤔 

I totally agree the deps you mentioned are not things we should just remove (like this one)

Reply:
The closest we have is https://github.com/esp-rs/esp-hal/issues/2525 with the `The PACs will probably never be 1.0, therefore we need to keep it out of the public API in esp-hal.` sentence. I don't see an issue linked or otherwise about <1.0 dependencies other than this.

Reply:
I'm still trying to understand what is wrong with marking unstable dependencies as unstable (rather than removing it).

Reply:
Let's discuss this outside this PR it moving forward, for now could you mark `uWrite` as an unstable impl again? (sorry)

Reply:
Done, thanks!